### ISSUE TITLE
Metaclasses using `Unpack[TypedDict]` for kwargs does not work as intended

### ISSUE THREAD

**Issue by mwath** [microsoft/pyright#7514]
**Describe the bug**
Creating a metaclass using `**kwargs: Unpack[MyTypedDict]` in its `__new__` method signature reports incorrect types when defining a new class inheriting the metaclass.

**Code or Screenshots**
```python
from typing import Any, TypedDict, Unpack


class Kwargs(TypedDict):
    bar: int


class FooMeta(type):
    def __new__(
        cls,
        clsname: str,
        bases: tuple[type, ...],
        classdict: dict[str, Any],
        **kwargs: Unpack[Kwargs],
    ) -> type: ...


class Foo(metaclass=FooMeta, bar=1): ...
#                                ^
# Argument of type "Literal[1]" cannot be assigned to parameter of type "*Kwargs" in function "__new__"
```

**Workarounds**
1. Using a `__init_subclass__` works as intended:
```python
class FooMeta:
    def __init_subclass__(cls, /, **kwargs: Unpack[Kwargs]) -> type: ...


class Foo(FooMeta, bar=1): ...  # Works as expected!
```
2. Defining the keyword arguments without using `Unpack` also works:
```python
class FooMeta(type):
    def __new__(
        cls,
        clsname: str,
        bases: tuple[type, ...],
        classdict: dict[str, Any],
        *,
        bar: int,
    ) -> type: ...


class Foo(metaclass=FooMeta, bar=1): ...  # Works
```

**VS Code extension or command-line**
The issue arises in both `pyright 1.1.354` from the command-line and `pylance v2024.3.100 (pre-release)` in VS Code

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
